leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -7,61 +8,404 @@
+                var seg2 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: '0011', startTP: 'A', startAT: myNull}); 
+                graph.getUserEvaluationBBM(this.cid, '0').set('assignedEvidencePair', '1100');
+            }
+            this.getFuncSegments().push(seg1, seg2);
+        }
+    }, 
+ 
+    /**
+     * Adds a new Constraint object int the global graph variable,
+     * representing an absolute constraint, if requried.
+     *
+     * If funcType is RC, CR, MP, MN, SD, DS  a Constraint object
+     * representing an absolute constraint will be added. If not, this
+     * function does not do anything
+     *
+     * @param {String} funcType
+     *   ex: 'RC'
+     */
+    addAbsConstraint: function(funcType) {
+
+        if (funcType == 'RC' || funcType == 'CR' || funcType == 'MP' ||
+            funcType == 'MN' || funcType == 'SD' || funcType == 'DS') {
+            graph.get('constraints').push(new ConstraintBBM({type: 'A', srcID: this.cid, srcRefTP: 'A', destID: null, destRefTP: null})); 
+        }
+    }, 
+ 
+    /**
+     * Removes the absolute Constraint object(s) for the Intention from
+     * the global graph variable, if such absolute Constraint object(s) exists
+     */
+
+    removeAbsConstraint: function() {
+        // TODO - unsure abt this function 
+
+        var i = 0;
+        while (i < graph.get('constraints').length) {
+            var constraint = graph.get('constraints');
+            if (constraint.get('type') == 'A' && constraint.get('srcID') === this.cid) {
+                graph.get('constraints').splice(i, 1); 
+            } else {
+                i++;
+            }
+        }
+    }, 
+ 
+    /**
+     * Adds a new FunctionSegmentBBM to the end of this Intention's EvolvingFunction's
+     * function list.
+ 
+     * Also adds a corresponding Constraint object reprenting an absolute constraint
+     * into the global graph variable.
+     *
+     *This function should only be used to add new function
+     * segments for user defined functions
+     *
+     * @param {String} funcType
+     *   ex: 'C'
+     * @param {String} refEvidencePair
+     *   ex: '0000'
+     * @param {Integer} startTime
+     */
+    addUserDefinedSeg: function(funcType, refEvidencePair, startTime){
+
+        var len = this.getFuncSegments().length;
+        var startCheck = this.getFuncSegments()[len - 1].get('startTP'); // Get last value in list 
+        if (startCheck == '0') { // If previous segment is at 0 then next one is at A
+            var start = 'A';
+        } else { // Otherwise, for following segment increase number by one 
+            var start = String.fromCharCode(startCheck.charCodeAt(0) + 1); 
+        }
+ 
+        // Create new FunctionSegmentBBM and adds it to functionSegList 
+        var new_model = new FunctionSegmentBBM({type: funcType, refEvidencePair: refEvidencePair, startTP: start, startAT: myNull}); 
+        this.getFuncSegments().push(new_model); 
```

Comment:
You can inline the new FunctionSegmentBBM line into the this.getFuncSegments().push() line because you don't use the variable anywhere else